disasterinfosite/load.py

```diff
@@ -1,14 +1,71 @@
+import math
 import os
+from django.contrib.gis.gdal import GDALRaster
+from django.contrib.gis.geos import Polygon
 from django.contrib.gis.utils import LayerMapping
 
 
+# If a raster's height or width is > this threshold, tile it
+# In theory this can be up to INT_MAX * 2, but that runs into memory constraints
+rasterTileMaxDimension = 5000
+
+
 ######################################################
 # GENERATED CODE GOES HERE
 # DO NOT MANUALLY EDIT CODE IN THIS SECTION - IT WILL BE OVERWRITTEN
 # loadMappings
 # END OF GENERATED CODE BLOCK
 ######################################################
 
+def tileLoadRaster(model, filename, band=0):
+    model.objects.all().delete()
```

Comment:
We don't delete old shapefiles on load - maybe we should?

Reply:
This actually clears the database rows.  I'm not 100% clear on why it needs to be done explicitly for a raster source and not a shapefile one, but my hypothesis is that there's a "unique on lookup_val" constraint somewhere on the geometries loaded from shapefiles and no equivalent for rasters.

Reply:
Sorry, that's what I meant - do you think it'd be a good idea to have them behave the same way? Right now, we just load more shapefile geometries in without deleting the old ones in that model. I wonder if that might make things weird, especially around the edges of geometries.

Reply:
When we load a shapefile, each lookup value becomes one row containing a MultiPolygon of all the areas for that value.  As far as I am aware, that overwrites any previous entry with the same lookup value in the same model.  I thought about implementing something similar for rasters but it would have to be keyed off tile numbers instead of lookup values, and that would then be prone to errors if we change the size or resolution of a raster or the value of `rasterTileMaxDimension` between loads.

Reply:
Oh right. Look at me forgetting how this thing that we made actually works...